src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -1002,7 +1008,8 @@ def do_resolve_with_resolved_children(
                 validation_mode=logical_plan.validation_mode,
                 column_names=logical_plan.column_names,
                 transformations=[
-                    self.analyze(x, df_aliased_col_name_to_real_col_name)
+                    # TODO investigate, this looks like a genuine type mismatch
```

Comment:
@sfc-gh-stan Looks like you introduced `df_aliased_col_name_to_real_col_name`. Could you take a look at the type mismatch here?

Reply:
The mismatch is from `CopyIntoTableNode.transformations`. Fixing...